src/App.tsx

```diff
@@ -1,61 +1,111 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // if there is no user with a given userId
+  return foundUser || null;
+}
+
+export const getTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState(getTodos);
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const formOnSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setTitleError(true);
+    }
+
+    if (!selectedUser) {
+      setUserError(true);
+    }
+
+    if (!selectedUser || !title.trim()) {
+      return;
+    }
+
+    const maxId = todos.map((todo) => (todo.id));
+    const newId = Math.max(...maxId) + 1;
+
+    const newTodo: Todo = {
+      id: newId,
+      completed: false,
+      user: getUser(Number(selectedUser)),
+      title: title.trim(),
+      userId: selectedUser,
+    };
+
+    setTodos([...todos, newTodo]);
```

Comment:
When you work with updating a state that already has some data, it is better to update it using a callback function, the first argument of which will be the actual current state. This will help avoid errors when rerendering the component.